tcpdevmem.c

```diff
@@ -10,6 +10,48 @@
 #include "thread.h"
 
 #define TEST_PREFIX "ncdevmem"
+#define RETURN_IF_NON_ZERO(cmd)	\
+	ret = (cmd);		\
+	if (ret) return ret;
+
+int driver_reset(const struct options *opts) {
+	char driver_reset_cmd[512];
+	int ret = 0;
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-strict-header-split on", opts->tcpd_link_name);
+	// ret = ret | system(driver_reset_cmd);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-strict-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-rxfh-indir %s equal 16", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool -K %s ntuple off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-strict-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool -K %s ntuple off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-max-rx-buffer-size on", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool -K %s ntuple on", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	printf("TCPDEVMEM driver reset returning %i\n", ret);
```

Comment:
You use printf here, but LOG_FATAL below. Please align on one correct way to log, either printf or LOG_FATAL/INFO/WARN, etc.

Reply:
This print is likely not necessary, it's better to print on error, and then not print if everything is OK, so that the logs are not spammy. If you really want to print, that's fine, but may LOG_INFO or LOG_DEBUG.

But this should always show 0 ret, becuase we return on non-zero ret.

Reply:
the intent was to signal in the console the status of cuda/udma buffer setup, but could be considered superfluous